File: CS50/sentimental-readability/readability.py
# Import get_string from cs50 library
from cs50 import get_string


def main():
    # Store users text input in a string
    text = get_string("Text: ")
    # Run functions to get numbers
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)
    L = float(letters / words * 100)
    S = float(sentences / words * 100)
    # Input values to get back Coleman Index and then print grade
    index = round(0.0588 * L - 0.296 * S - 15.8)
    index = int(index)
    print_value(index)

# Create a function to count words


def count_words(string):
    count = 1
    for i in string:
        if i.isspace():
            count += 1
    return count

# Create a function to count letters


def count_letters(string):
    count = 0
    for i in string:
        if i.isalpha():
            count += 1
    return count

# Create a function to count sentences


def count_sentences(string):
    count = 0
    for i in string:
        if i == "." or i == "?" or i == "!":
            count += 1
    return count

# Print result based on value


def print_value(index):
    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


# Run Main
main()

File: CS50/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int countLetters(string s);
int countWords(string s);
int countSentences(string s);

int main(void)
{
    string userInput = get_string("Text: ");
    // Store all the results of count functions as floats to handle the fun math
    float letters = countLetters(userInput);
    float words = countWords(userInput);
    float sentences = countSentences(userInput);
    // Letters per 100 words
    float L = letters / words * 100;
    // Sentences per 100 words
    float S = sentences / words * 100;
    // Plug all that fun into the Coleman-Liau index and print out based on the value.
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int countLetters(string s)
{
    int length = strlen(s);
    int letters = 0;
    //Loop through every element of an array searching for letters only, upper and lower.
    for (int i = 0; i < length; i++)
    {
        if (isalpha(s[i]))
        {
            letters += 1;
        }
    }
    return letters;
}

int countWords(string s)
{
    //Int for holding string length
    int length = strlen(s);
    int words = 0;
    //Loop through every element of an array searching for spaces and punctuation
    for (int i = 0; i < length; i++)
    {
        if (isspace(s[i]))
        {
            words += 1;
        }
        //If there is punctuation, and then a space afterwards, add one to word.
        else if ((s[i] == 33 && (s[i + 1] == 0))  || (s[i] == 46 && (s[i + 1] == 0)) || (s[i] == 63 && (s[i + 1] == 0)))
        {
            words += 1;
        }
    }
    return words;
}

int countSentences(string s)
{
    int length = strlen(s);
    int sentences = 0;
    //Loop through every element of an array searching for punctuation to end a sentence.
    for (int i = 0; i < length; i++)
    {
        if (s[i] == 33 || s[i] == 46 || s[i] == 63)
        {
            sentences += 1;
        }
    }
    return sentences;
}